pkg/engine/variables/operator/allin.go

```diff
@@ -115,14 +119,16 @@ func isAllIn(key []string, value []string) bool {
 
 // isAllNotIn checks if all the values in S1 are not in S2
 func isAllNotIn(key []string, value []string) bool {
+	found := 0
 	for _, valKey := range key {
 		for _, valValue := range value {
 			if wildcard.Match(valKey, valValue) {
-				return false
+				found++
```

Comment:
I feel like this can still have unintended results if `value` has duplicates. 
E.g.:
```{kyverno.Condition{Key: []interface{}{"1.1.1.1", "4.4.4.4"}, Operator: kyverno.AllNotIn, Value: []interface{}{"1.1.1.1", "1.1.1.1", "2.2.2.2", "3.3.3.3"}}, true},```

Reply:
I think 
```
found++
continue
```
might work but not certain without testing

Reply:
Hey @MarcelMue thank you your review,
I can also see this happening,
I will try to make some changes here today

Reply:
@MarcelMueI have made some corrections for this.